cbc-repeated-iv/src/cbc_game.rs

```diff
@@ -0,0 +1,528 @@
+use aes::cipher::block_padding::UnpadError;
+use aes::cipher::{block_padding::Pkcs7, BlockDecryptMut, BlockEncryptMut, KeyIvInit};
+use colored::Color::TrueColor;
+use colored::{Color, Colorize};
+use std::fmt::Write;
+
+type Aes128CbcEnc = cbc::Encryptor<aes::Aes128>;
+type Aes128CbcDec = cbc::Decryptor<aes::Aes128>;
+
+use rand::Rng;
+
+const BLOCK_SIZE: usize = 16;
+
+pub struct CBCGame {
+    key: [u8; BLOCK_SIZE],
+    iv: [u8; BLOCK_SIZE],
+
+    known_plaintext: Vec<u8>,
+    known_ciphertext: Vec<u8>,
+
+    plaintexts_vec: Vec<Vec<u8>>,
+    ciphertexts_vec: Vec<Vec<u8>>,
+
+    cbc_same_iv_ascii_art: String,
+}
+
+impl Default for CBCGame {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl CBCGame {
+    pub fn new() -> Self {
+        let mut rng = rand::thread_rng();
+        Self {
+            key: rng.gen(),
+            iv: rng.gen(),
+
+            known_plaintext: br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "S"}"#
+                .to_vec(),
+            known_ciphertext: vec![],
+
+            plaintexts_vec: vec![
+                br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "S"}"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "M"}"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Tom",  "Surname": "S"}"#.to_vec(),
+                br#"{  "UserType": "Administrator", "Name": "Alex", "Surname": "S"}"#.to_vec(),
+                br#"{ "Heartbeat": "Alive",         "Since": "1653924357" }"#.to_vec(),
+            ],
+            ciphertexts_vec: vec![],
+
+            cbc_same_iv_ascii_art: "
+
+     ▄████████ ▀█████████▄   ▄████████         ▄████████    ▄████████   ▄▄▄▄███▄▄▄▄      ▄████████       ▄█   ▄█    █▄
+    ███    ███   ███    ███ ███    ███        ███    ███   ███    ███ ▄██▀▀▀███▀▀▀██▄   ███    ███      ███  ███    ███
+    ███    █▀    ███    ███ ███    █▀         ███    █▀    ███    ███ ███   ███   ███   ███    █▀       ███▌ ███    ███
+    ███         ▄███▄▄▄██▀  ███               ███          ███    ███ ███   ███   ███  ▄███▄▄▄          ███▌ ███    ███
+    ███        ▀▀███▀▀▀██▄  ███             ▀███████████ ▀███████████ ███   ███   ███ ▀▀███▀▀▀          ███▌ ███    ███
+    ███    █▄    ███    ██▄ ███    █▄                ███   ███    ███ ███   ███   ███   ███    █▄       ███  ███    ███
+    ███    ███   ███    ███ ███    ███         ▄█    ███   ███    ███ ███   ███   ███   ███    ███      ███  ███    ███
+    ████████▀  ▄█████████▀  ████████▀        ▄████████▀    ███    █▀   ▀█   ███   █▀    ██████████      █▀    ▀██████▀
+
+".to_string()
+        }
+    }
+
+    pub fn start(&mut self) -> anyhow::Result<()> {
+        let mut iter = 0;
+        let mut line = String::new();
+
+        println!("{}", self.cbc_same_iv_ascii_art.on_color(Color::Green));
+
+        self.reinitialize_enc_dec();
```

Comment:
We already initialize the key and iv. Seems like a leftover :)

Reply:
If you want, sure it can. But in this case I think this method should call `encrypt_all_texts`, so the object will be in consistent state.